cmd/codecrafters/main.go

```diff
@@ -76,7 +81,11 @@ func run() error {
 
 	switch cmd {
 	case "test":
-		return commands.TestCommand(ctx)
+		testCmd := flag.NewFlagSet("test", flag.ExitOnError)
+		previous := testCmd.Bool("previous", false, "run tests for the current stage and all previous stages in ascending order")
+		testCmd.Parse(flag.Args()[1:]) // parse the args after the test command
```

Comment:
Just curious - what happens if a user makes a typo here? If they do `codecrafters test --previos` for example? Will this complain? (Ideally it should)

Reply:
<img width="1558" alt="Screenshot 2024-06-25 at 6 37 02 PM" src="https://github.com/codecrafters-io/cli/assets/62695092/c382097b-bb60-4c04-ae5b-4fa480f21155">

tried it out and it does complain if a nonexistent flag is used for test